src/main/java/com/avispl/symphony/dal/infrastructure/management/samsung/smartthings/SamsungSmartThingsAggregatorCommunicator.java

```diff
@@ -0,0 +1,3177 @@
+			}
+		} catch (Exception e) {
+			failedMonitoringDeviceIds.add(deviceId);
+			logger.error(String.format("Error while retrieve %s full controllable properties status info: %s", cachedDevicesAfterPollingInterval.get(deviceId).getName(), e.getMessage()), e);
+		}
+	}
+
+	/**
+	 * This method is used to map controllable property status to device
+	 *
+	 * @param response Json response
+	 * @param detailViewPresentations list of detail view presentation of device
+	 */
+	private List<DetailViewPresentation> mapControllablePropertyStatusToDevice(ObjectNode response, List<DetailViewPresentation> detailViewPresentations, DevicePresentation devicePresentation) {
+
+		for (DetailViewPresentation detailViewPresentation : detailViewPresentations) {
+
+			Optional<Iterator<JsonNode>> controllablePropertiesStatus = Optional.ofNullable(response.elements().next())
+					.map(JsonNode::elements)
+					.map(Iterator::next)
+					.map(c -> c.get(detailViewPresentation.getCapability()))
+					.map(JsonNode::elements);
+
+			String value = SmartThingsConstant.NONE;
+			String unit = SmartThingsConstant.NONE;
+
+			// map specific controllable property status to device
+			switch (detailViewPresentation.getCapability()) {
+				case AggregatedDeviceColorControllingConstant.COLOR_CONTROL:
+					if (detailViewPresentation.getDisplayType().equals(DeviceDisplayTypesMetric.NUMBER_FIELD)) {
+						if (AggregatedDeviceColorControllingConstant.COLOR_CONTROL_SET_HUE.equals(detailViewPresentation.getNumberField().getCommand())) {
+							value = Optional.ofNullable(response.elements().next())
+									.map(JsonNode::elements)
+									.map(Iterator::next)
+									.map(c -> c.get(detailViewPresentation.getCapability()))
+									.map(d -> d.get(AggregatedDeviceColorControllingConstant.COLOR_CONTROL_HUE))
+									.map(u -> u.get(SmartThingsConstant.VALUE))
+									.map(JsonNode::asText)
+									.orElse(SmartThingsConstant.NONE);
+							devicePresentation.getColor().setHue(Float.parseFloat(value));
+							continue;
+						}
+						if (AggregatedDeviceColorControllingConstant.COLOR_CONTROL_SET_SATURATION.equals(detailViewPresentation.getNumberField().getCommand())) {
+							value = Optional.ofNullable(response.elements().next())
+									.map(JsonNode::elements)
+									.map(Iterator::next)
+									.map(c -> c.get(detailViewPresentation.getCapability()))
+									.map(d -> d.get(AggregatedDeviceColorControllingConstant.COLOR_CONTROL_SATURATION))
+									.map(u -> u.get(SmartThingsConstant.VALUE))
+									.map(JsonNode::asText)
+									.orElse(SmartThingsConstant.NONE);
+							devicePresentation.getColor().setSaturation(Float.parseFloat(value));
+							continue;
+						}
+					}
+					break;
+				default:
+					break;
```

Comment:
the question above + a log entry here, indicating that there's no expected behavior for this exist

Reply:
I used to map for some unique controllable properties. if it's common control, we'll map it in the next step. I'm going to take a note here "we do not log error/ warning here, we are going to map data for common controllable properties in the next step"